analytical_engine/apps/java_pie/java_pie_projected_parallel_app.h

```diff
@@ -0,0 +1,125 @@
+/** Copyright 2020 Alibaba Group Holding Limited.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef ANALYTICAL_ENGINE_APPS_JAVA_PIE_JAVA_PIE_PROJECTED_PARALLEL_APP_H_
+#define ANALYTICAL_ENGINE_APPS_JAVA_PIE_JAVA_PIE_PROJECTED_PARALLEL_APP_H_
+
+#ifdef ENABLE_JAVA_SDK
+
+#include <utility>
+
+#include "grape/communication/communicator.h"
+#include "grape/grape.h"
+#include "grape/types.h"
+
+#include "core/context/java_pie_projected_context.h"
+#include "core/error.h"
+
+namespace gs {
+
+/**
+ * @brief This is a driver app for Java app. The driven java app should be
+ * inherited from ProjectedDefaultAppBase.
+ *
+ * @tparam FRAG_T Should be gs::ArrowProjectedFragment<...>
+ */
+template <typename FRAG_T>
+class JavaPIEProjectedParallelApp
```

Comment:
Could you make the naming unified?
Projected? or use the Simple?

Reply:
Do you mean `class JavaSimpleParallelApp`? If you mean changing the naming to `class JavaSimpleParallelApp`, I'm ok on that.

If you are asking whether we can unify parallelApp and defaultApp as one, I'm afraid that we need to separate file for parallel and default app, since its method signatures are different.

Reply:
OK, it is fine to keep the current name.